src/plugins/intel_gpu/src/graph/arg_max_min.cpp

```diff
@@ -136,30 +102,61 @@ layout arg_max_min_inst::calc_output_layout(arg_max_min_node const& node) {
                                      input_layout.spatial(0),
                                      input_layout.spatial(1),
                                      (int32_t)desc->top_k}};
-            default:
-                break;
-        }
+            } else {
+                size_check(input_layout.spatial(1));
+                return layout{output_data_type,
+                              format,
+                              tensor{input_layout.batch(),
+                                     input_layout.feature(),
+                                     input_layout.spatial(0),
+                                     (int32_t)desc->top_k}};
+            }
+        case 3:
+            size_check(input_layout.spatial(1));
+            if (format == cldnn::format::bfzyx) {
+                return layout{output_data_type,
+                              format::bfzyx,
+                              tensor{input_layout.batch(),
+                                     input_layout.feature(),
+                                     input_layout.spatial(0),
+                                     (int32_t)desc->top_k,
+                                     input_layout.spatial(2)}};
+            } else {
+                size_check(input_layout.spatial(0));
+                return layout{output_data_type,
+                              format,
+                              tensor{input_layout.batch(),
+                                     input_layout.feature(),
+                                     (int32_t)desc->top_k,
+                                     input_layout.spatial(1)}};
+            }
+        case 4:
+            size_check(input_layout.spatial(0));
+            return layout{output_data_type,
+                          format::bfzyx,
+                          tensor{input_layout.batch(),
+                                 input_layout.feature(),
+                                 (int32_t)desc->top_k,
+                                 input_layout.spatial(1),
+                                 input_layout.spatial(2)}};
+        default:
+            IE_THROW() << "Incorrect arg_max_min axis.";
     }
-    size_check(input_layout.feature() * input_layout.spatial(0) * input_layout.spatial(1));
-    return layout{output_data_type,
-                  input_layout.format,
-                  tensor{input_layout.batch(), 1, (int32_t)desc->top_k, 1}};
 }
 
 std::string arg_max_min_inst::to_string(arg_max_min_node const& node) {
     auto desc = node.get_primitive();
     auto node_info = node.desc_to_json();
-    auto axis = desc->with_axis ? "true" : "false";
-    auto out_type = desc->output_type ? "max" : "min";
+    auto out_type = desc->mode == ov::op::TopKMode::MAX ? "max" : "min";
+    auto sort_type = desc->sort == ov::op::TopKSortType::SORT_INDICES ? "sort_indices" : "sort_values";
```

Comment:
Both ngraph enums have operator<< defined. So please use it instead of self defined conversions to string.